firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigSharedPrefsClient.java

```diff
@@ -249,6 +257,46 @@ void setBackoffMetadata(int numFailedFetches, Date backoffEndTime) {
     }
   }
 
+  public void setCustomSignals(Map<String, Object> newCustomSignals) {
+    synchronized (customSignalsLock) {
+      // Retrieve existing custom signals
+      Map<String, Object> existingCustomSignals = getCustomSignals();
+
+      for (Map.Entry<String, Object> entry : newCustomSignals.entrySet()) {
```

Comment:
do we have any limits on how many custom signals can be set? or limits on the key/value size?

Reply:
Yes, following our recent discussions, I've added limits to custom signals: a maximum of 100 signals, with key sizes up to 250 characters and value sizes up to 500 characters.